src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -264,13 +278,38 @@ private DAGRequest buildTableScan() {
     return request;
   }
 
+  private void setColumnOffsets(TiExpr expr) {
```

Comment:
by seeing column offset, tikv can take advantage of it. But my concern is where is the logic in tidb side, I just want to make sure this logic can do the trick.

Reply:
This may lie in TiDB's `schema`(may hard to find in source code...) or TiKV-Client developers' document.